### PR TITLE
NMS-16482: Version in about page points to invalid release notes.

### PR DESCRIPTION
**Pull request by Junaid-Khan-Nant**
- https://opennms.atlassian.net/browse/NMS-16482

### CODE REVIEW

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 1]
```diff
@@ -70,7 +70,7 @@
 <table class="table table-sm">
   <tr>
     <th>Version:</th>
-    <td><a href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("version.display")%></a></td>
+    <td><a href="https://docs.opennms.com/meridian/<%="2024"%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("version.display")%></a></td>
```
[the remark this answers is not in the document]
Comment (smunir-onms) [reply opennms-webapp/src/main/webapp/about/index.jsp:73]: Right, to make it dynamic there should be a property like "opennms.product" and the version property shouldn't be replaced with hardcoded "value 2024"

**Review by indigo423 — CHANGES_REQUESTED** [review]
This change needs to be made in the private repository https://github.com/opennms/opennms-prime. @cgorantla, yes this is a good idea having a property where we can set the "distribution" at compile time would be helpful.

**Comment by Junaid-Khan-Nant** [comment]
> This change needs to be made in the private repository https://github.com/opennms/opennms-prime. @cgorantla, yes this is a good idea having a property where we can set the "opennms.product" at compile time would be helpful.

@cgorantla , don't have access for this [repository.](https://github.com/opennms/opennms-prime).

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 2]
```diff
@@ -70,7 +70,7 @@
 <table class="table table-sm">
   <tr>
     <th>Version:</th>
-    <td><a href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("version.display")%></a></td>
+    <td><a href="https://docs.opennms.com/<%=Vault.getProperty("opennms.product")%>/<%=Vault.getProperty("release.version")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("opennms.product")%> - <%=Vault.getProperty("release.version")%></a></td>
```
Comment (smunir-onms) [inline opennms-webapp/src/main/webapp/about/index.jsp:73]: Don't add a new property "release.version", use "version.display" instead, docs page path and properties values would be adjusted accordingly. @cgorantla @indigo423 please correct me if am wrong. 
```suggestion 
    <td><a href="https://docs.opennms.com/<%=Vault.getProperty("opennms.product")%>/<%=Vault.getProperty("release.version")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("opennms.product")%> - <%=Vault.getProperty("version.display")%></a></td>
```

File: opennms-webapp/src/main/web-filtered/WEB-INF/version.properties [hunk 3]
```diff
@@ -26,6 +26,5 @@
 #     http://www.opennms.com/
 ###############################################################################
 
-version.display=${display.version}
-opennms.product=meridian
-release.version=2024
\ No newline at end of file
+version.display=2024.1.3
```
Comment (indigo423) [inline opennms-webapp/src/main/web-filtered/WEB-INF/version.properties:29]: We should not hardcode the version here. The `${display.version}` is set from the pom.xml using `${install.version}` and follows automatically the correct version number. Hardcoding the version number would require dealing with it manually.

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 4]
```diff
@@ -70,7 +70,7 @@
 <table class="table table-sm">
   <tr>
     <th>Version:</th>
-    <td><a href="https://docs.opennms.com/<%=Vault.getProperty("opennms.product")%>/<%=Vault.getProperty("release.version")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("opennms.product")%> - <%=Vault.getProperty("release.version")%></a></td>
+    <td><a href="https://docs.opennms.com/<%=Vault.getProperty("opennms.product")%>/<%=Vault.getProperty("version.display")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("opennms.product")%> - <%=Vault.getProperty("version.display")%></a></td>
```
Comment (indigo423) [inline opennms-webapp/src/main/webapp/about/index.jsp:73]: What about doing a similar thing, setting a property in the pom.xml like `${install.product}` which is evaluated from a single place from the pom.xml which we can control at build time with maven?

**Review by indigo423 — CHANGES_REQUESTED** [review]
We should think about a way to control these settings at build time with maven. It allows us to set it in the build jobs for Meridian / Horizon rather than hard coding it in properties.

**Comment by Junaid-Khan-Nant** [comment]
Hi @indigo423,  the property related to product name has been added and tested, and it's working fine. I am facing issues on the version as it is also used in inner pom's xml. If I replace this 33.0.3-SNAPSHOT with 2024.1.3-SNAPSHOT, then initial is working fine, but after a few times it stops the server with the following logs.
log-2
log-1
Any idea about it? Or I believe there is another way to create a new tag just like we created for the product and mention the release version there.It would be in parent pom.xml.Your opinion?

File: pom.xml [hunk 5]
```diff
@@ -1668,6 +1668,8 @@
     <mock.db.adminPassword/>
     <mock.runSnmpTests>false</mock.runSnmpTests>
     <mock.snmpHost>localhost</mock.snmpHost>
+    <install.product>meridian</install.product>
```
Comment (mershad-manesh) [inline pom.xml:1671]: This should be `horizon` as `Foundation-` branches are based on `horizon`.

File: pom.xml [hunk 6]
```diff
@@ -1668,6 +1668,8 @@
...
+    <product.name>${install.product}</product.name>
```
Comment (cgorantla) [inline pom.xml:1672]:
```suggestion
    <product.name>meridian</product.name>
```

File: pom.xml (hunk 5 above)
Comment (cgorantla) [inline pom.xml:1671]:
```suggestion
```

File: pom.xml (hunk 6 above)
Comment (cgorantla) [reply pom.xml:1672]: Add another property as
<docs.product.version>24</docs.product.version>

**Review by cgorantla — COMMENTED** [review]
Also you can update this in following pages

````
git grep "docs.opennms.com"
src/main/webapp/about/index.jsp:73:    <td><a href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/releasenotes/whatsnew.html" target="_blank" title="Release Notes"><%=Vault.getProperty("version.display")%></a></td>
src/main/webapp/includes/help-documentation.jsp:63:                            href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/deployment/core/introduction.html"
src/main/webapp/includes/help-documentation.jsp:70:                            href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/operation/overview/overview.html#overview"
src/main/webapp/includes/help-documentation.jsp:77:                            href="https://docs.opennms.com/horizon/<%=Vault.getProperty("version.display")%>/development/overview/overview.html#overview"
```

File: pom.xml [hunk 7]
```diff
@@ -1668,6 +1668,8 @@
     <mock.db.adminPassword/>
     <mock.runSnmpTests>false</mock.runSnmpTests>
     <mock.snmpHost>localhost</mock.snmpHost>
+    <product.name>meridian</product.name>
+    <docs.product.version>2024</docs.product.version>
```
Comment (cgorantla) [inline pom.xml:1672]:
```suggestion
    <docs.product.version>33</docs.product.version>
```

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**